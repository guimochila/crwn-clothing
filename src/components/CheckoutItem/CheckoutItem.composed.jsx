import React from 'react';
import { compose, graphql } from 'react-apollo';

import CheckoutItem from './CheckoutItem';
import {
  ADD_ITEM_TO_CART,
  REMOVE_ITEM_FROM_CART,
  CLEAR_ITEM_FROM_CART,
} from '../../graphql/queries';

const CheckoutItemComposed = ({
  addItemToCart,
  removeItemFromCart,
  clearItemFromCart,
  ...otherProps
}) => (
  <CheckoutItem
    {...otherProps}
    addItem={item => addItemToCart({ variables: { item } })}
    removeItem={item => removeItemFromCart({ variables: { item } })}
    clearItem={item => clearItemFromCart({ variables: { item } })}
  />
);

export default compose(
  graphql(ADD_ITEM_TO_CART, { name: 'addItemToCart' }),
  graphql(REMOVE_ITEM_FROM_CART, { name: 'removeItemFromCart' }),
  graphql(CLEAR_ITEM_FROM_CART, { name: 'clearItemFromCart' }),
)(CheckoutItemComposed);
