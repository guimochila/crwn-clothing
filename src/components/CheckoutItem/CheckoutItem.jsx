import React from 'react';

import {
  CheckoutItemContainer,
  ImageContainer,
  QuantityContainer,
  TextContainer,
  RemoveButtonContainer,
} from './CheckoutItem.styles';

function CheckoutItem({ cartItem, clearItem, addItem, removeItem }) {
  const { name, imageUrl, price, quantity } = cartItem;

  return (
    <CheckoutItemContainer>
      <ImageContainer>
        <img src={imageUrl} alt="item" />
      </ImageContainer>
      <TextContainer>{name}</TextContainer>
      <QuantityContainer>
        <div onClick={() => removeItem(cartItem)}>&#10094;</div>
        <span>{quantity}</span>
        <div onClick={() => addItem(cartItem)}>&#10095;</div>
      </QuantityContainer>
      <TextContainer>{price}</TextContainer>
      <RemoveButtonContainer onClick={() => clearItem(cartItem)}>
        &#10005;
      </RemoveButtonContainer>
    </CheckoutItemContainer>
  );
}

export default CheckoutItem;
