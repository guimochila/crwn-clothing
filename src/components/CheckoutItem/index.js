import CheckoutItemComposed from './CheckoutItem.composed';

export default CheckoutItemComposed;
