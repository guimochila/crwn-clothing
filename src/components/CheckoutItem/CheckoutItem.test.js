import React from 'react';
import { shallow } from 'enzyme';

import { CheckoutItem } from './CheckoutItem';

describe('CheckoutItem component', () => {
  let wrapper;
  const mockCartItem = {
    name: 'Brown Hat',
    imageUrl: 'https://imgsrc.com/brownhat.jpg',
    price: 100,
    quantity: 2,
  };
  const clearItem = jest.fn();
  const addItem = jest.fn();
  const removeItem = jest.fn();

  beforeEach(() => {
    const mockProps = {
      cartItem: mockCartItem,
      clearItem,
      addItem,
      removeItem,
    };

    wrapper = shallow(<CheckoutItem {...mockProps} />);
  });

  test('it should render CheckoutItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it should call removeItem when click in remove', () => {
    wrapper.find('[data-testid="removeItem"]').simulate('click');
    expect(removeItem).toHaveBeenCalledWith(mockCartItem);
  });

  test('it should call addItem when click in add', () => {
    wrapper.find('[data-testid="addItem"]').simulate('click');
    expect(addItem).toHaveBeenCalledWith(mockCartItem);
  });

  test('it should call clearItem when click in the clear button', () => {
    wrapper.find('[data-testid="clearItem"]').simulate('click');
    expect(clearItem).toHaveBeenLastCalledWith(mockCartItem);
  });
});
