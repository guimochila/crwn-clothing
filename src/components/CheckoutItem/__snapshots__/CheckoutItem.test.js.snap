// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component it should render CheckoutItem component 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="item"
      src="https://imgsrc.com/brownhat.jpg"
    />
  </styled.div>
  <styled.span>
    Brown Hat
  </styled.span>
  <Styled(styled.span)>
    <div
      data-testid="removeItem"
      onClick={[Function]}
    >
      ❮
    </div>
    <span>
      2
    </span>
    <div
      data-testid="addItem"
      onClick={[Function]}
    >
      ❯
    </div>
  </Styled(styled.span)>
  <styled.span>
    100
  </styled.span>
  <styled.div
    data-testid="clearItem"
    onClick={[Function]}
  >
    ✕
  </styled.div>
</styled.div>
`;
